libs/lua-hash/src/hash.c

```diff
@@ -0,0 +1,47 @@
+#include <libubox/md5.h>
+
+#include <lua.h>
+#include <lauxlib.h>
+#include <lualib.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define MD5_DIGEST_LEN 16
+#define MD5_STRLEN 2*MD5_DIGEST_LEN+1
+
+static int md5(lua_State *L) {
+	char *input = (char *) lua_tostring(L, 1);
+	if (input == NULL) {
+		lua_pushstring(L, "first argument: should be the input string");
+		lua_error(L);
+		return 1;
+	}
+
+	char digest[MD5_DIGEST_LEN];
+	char output[MD5_STRLEN];
+
+	struct md5_ctx ctx;
+	md5_begin(&ctx);
+	md5_hash(input, strlen(input), &ctx);
+	md5_end(digest, &ctx);
+
+	// fill the digest bytewise in the output string
+	int i;
+	for (i = 0; i < MD5_DIGEST_LEN; i++) {
+		sprintf(output + 2*i, "%02x", (unsigned char) digest[i]);
```

Comment:
Instead of casting to unsigned char here, you could just declare *digest* as an unsigned char array.